src/mlpack/core/data/load_csv_impl.hpp

```diff
@@ -0,0 +1,222 @@
+/** 
+ * @file core/data/load_csv_impl.hpp
+ * 
+ * @author Conrad Sanderson
+ * @author Gopi M. Tatiraju
+ *
+ * This csv parser is designed by taking reference from armadillo's csv parser.
+ * In this mlpack's version, all the arma dependencies were removed or replaced
+ * accordingly, making the parser totally independent of armadillo.
+ *
+ * This parser will be totally independent to any linear algebra library.
+ * This can be used to load data into any matrix, i.e. arma and bandicoot
+ * in future.
+ *
+ * https://gitlab.com/conradsnicta/armadillo-code/-/blob/10.5.x/include/armadillo_bits/diskio_meat.hpp
+ * Copyright 2008-2016 Conrad Sanderson (http://conradsanderson.id.au)
+ * Copyright 2008-2016 National ICT Australia (NICTA)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_CSV_PARSER_IMPL_HPP
+#define MLPACK_CORE_DATA_CSV_PARSER_IMPL_HPP
+
+#include "load_csv.hpp"
+
+namespace mlpack
+{
+namespace data
+{
+  /**
+   * Given the address of a martix element(val)
+   * sets it equal to the provided value(token)
+   * example calling: convert_token<eT>(x.at(row, col), token)
```

Comment:
change `convert_token<eT>(...)` to `LoadCSV::ConvertToken(...)`

Reply:
Oh, we don't need it as a template here I guess, sounds good. Thank you for the review.